Commit for push

TheBrockstar opened the pull request:
# Description

Adds backend functionality for sending messages. Please ignore the single commit. I lost all of them at some point.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change status
- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?

- [x] Test A: Tested with missing information.
- [x] Test B: Tested with correct information.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

TheBrockstar committed:
Commit for push

easyra reviewed (dismissed):
Endpoint seems to be working for me

easyra reviewed (approved):
The endpoint is working perfectly for me

TheBrockstar committed:
Update Readme

TheBrockstar committed:
Separate UIDs in CID With Underscore
- The CID is join of the two participants uids.
- The join delimiter is now an underscore.

TheBrockstar committed:
Add Comments to Clarify User/Participant
- User is the sender.
- Participant is the receiver.
